controllers/commonservice_controller.go

```diff
@@ -70,22 +79,45 @@ func (r *CommonServiceReconciler) ReconcileMasterCR(req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if instance.Status.Phase == "" {
+		if err := r.updatePhase(instance, CRInitializing); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		if err := r.updatePhase(instance, CRUpdating); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
 	// Init common service bootstrap resource
 	// Including namespace-scope configmap, nss operator, nss CR
 	// Webhook Operator and Secretshare
 	// Delete ODLM from openshift-operators and deploy it in the masterNamespaces
 	// Deploy OperandConfig and OperandRegistry
 	if err := r.Bootstrap.InitResources(instance.Spec.ManualManagement); err != nil {
 		klog.Errorf("Failed to initialize resources: %v", err)
+		if err := r.updatePhase(instance, CRFailed); err != nil {
+			klog.Error(err)
+		}
 		return ctrl.Result{}, err
 	}
 
 	newConfigs, err := r.getNewConfigs(req)
 	if err != nil {
+		if err := r.updatePhase(instance, CRFailed); err != nil {
+			klog.Error(err)
+		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
Did the new err form `updatePhase` override the previous err from `getNewConfigs` in the return?
Would we expect this situation since the first err is not reported in log?

Reply:
@Daniel-Fan 
The `err` in the `if` condition wouldn't override the outside `err`. It is a local variable within the if logic.

The out layer `err` is recorded in the reconcile logs.

Reply:
Get it!
That's a new point I haven't noticed before, thank you!